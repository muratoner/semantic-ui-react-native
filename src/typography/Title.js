import PropTypes from 'prop-types';
import React, { useMemo } from 'react';
import { StyleSheet, Text, ViewPropTypes } from 'react-native';
import { TitleProps } from '../';

const Component = (props: TitleProps) => {
	const { style, size, ...attributes } = props;

	const fontSize = useMemo(() => {
		switch (size) {
		case 'huge':
			return 28;
		case 'large':
			return 24;
		case 'small':
			return 15;
		case 'tiny':
			return 14;
		case 'medium':
		default:
			return 18;
		}
	}, [size]);

	return <Text {...attributes} style={[{ fontSize }, styles.title, style]} />;
};

const styles = StyleSheet.create({
	title: {
		fontWeight: 'bold'
	}
});

Component.protoTypes = {
	...ViewPropTypes,
	text: PropTypes.string,
	type: PropTypes.oneOf(['huge', 'large', 'medium', 'small', 'tiny'])
};

export default Component;
