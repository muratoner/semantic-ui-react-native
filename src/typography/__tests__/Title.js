import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Text, View } from 'react-native';
import Title from '../Title';

describe('Title Component', () => {
	it('should render without issues', () => {
		const component = shallow(<Title>Title Test 1</Title>);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with children', () => {
		const component = shallow(
			<Title>
				<View>
					<Text>Title Test 2</Text>
				</View>
			</Title>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with style', () => {
		const component = shallow(
			<Title style={{ color: '#191919' }}>Title Test 1</Title>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with size huge', () => {
		const component = shallow(<Title size="huge">Title Test 1</Title>);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with size large', () => {
		const component = shallow(<Title size="large">Title Test 1</Title>);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with size medium', () => {
		const component = shallow(<Title size="medium">Title Test 1</Title>);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with size small', () => {
		const component = shallow(<Title size="small">Title Test 1</Title>);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with size tiny', () => {
		const component = shallow(<Title size="tiny">Title Test 1</Title>);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});
});
