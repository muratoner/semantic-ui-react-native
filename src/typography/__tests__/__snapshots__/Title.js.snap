// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component should render with children 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 18,
      },
      Object {
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  <View>
    <Text>
      Title Test 2
    </Text>
  </View>
</Text>
`;

exports[`Title Component should render with size huge 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 28,
        "fontWeight": "bold",
      },
      Object {
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  Title Test 1
</Text>
`;

exports[`Title Component should render with size large 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
      },
      Object {
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  Title Test 1
</Text>
`;

exports[`Title Component should render with size medium 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
      },
      Object {
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  Title Test 1
</Text>
`;

exports[`Title Component should render with size small 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 15,
        "fontWeight": "bold",
      },
      Object {
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  Title Test 1
</Text>
`;

exports[`Title Component should render with size tiny 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 14,
        "fontWeight": "bold",
      },
      Object {
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  Title Test 1
</Text>
`;

exports[`Title Component should render with style 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#191919",
        "fontSize": 18,
        "fontWeight": "bold",
      },
      Object {
        "fontWeight": "bold",
      },
      Object {
        "color": "#191919",
      },
    ]
  }
>
  Title Test 1
</Text>
`;

exports[`Title Component should render without issues 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 18,
      },
      Object {
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  Title Test 1
</Text>
`;
